internal/status/kv.go

```diff
@@ -68,23 +67,101 @@ func WatchConditionStatus(ctx context.Context, kind ptypes.ConditionKind, facili
 	if !ok {
 		return nil, errors.Wrap(errNoKV, string(kind))
 	}
+	return bucket, nil
+}
+
+// WatchConditionStatus specializes some generic NATS functionality, mainly to keep
+// the callers cleaner of the NATS-specific details.
+func WatchConditionStatus(ctx context.Context, kind ptypes.ConditionKind, facility string) (nats.KeyWatcher, error) {
+	bucket, err := getKVBucket(kind)
+	if err != nil {
+		return nil, err
+	}
 
 	// format the facility as a NATS subject to use as a filter for relevant KVs
 	keyStr := fmt.Sprintf("%s.*", facility)
-	return bucket.Watch(keyStr, nats.Context(ctx))
+	return bucket.Watch(keyStr, nats.Context(ctx), nats.IgnoreDeletes())
 }
 
 // GetConditionKV returns the raw NATS KeyValue interface for the bucket associated
-// with the given condition type.
+// with the given condition type. This is a really low-level access, but if you want
+// a handle to the raw NATS API, here it is.
 func GetConditionKV(kind ptypes.ConditionKind) (nats.KeyValue, error) {
-	if !kvReady {
-		return nil, errNotReady
+	bucket, err := getKVBucket(kind)
+	if err != nil {
+		return nil, err
 	}
 
-	bucket, ok := kvCollection[string(kind)]
-	if !ok {
-		return nil, errors.Wrap(errNoKV, string(kind))
+	return bucket, nil
+}
+
+// DeleteCondition does what it says on the tin. If this does not return an error, the
+// KV entry is gone.
+func DeleteCondition(kind ptypes.ConditionKind, facility, condID string) error {
+	bucket, err := getKVBucket(kind)
+	if err != nil {
+		return err
 	}
 
-	return bucket, nil
+	return bucket.Delete(facility + "." + condID)
+}
+
+// GetSingleCondition does exactly that given a kind, facility, and condition-id
+func GetSingleCondition(kind ptypes.ConditionKind, facility, condID string) (nats.KeyValueEntry, error) {
+	bucket, err := getKVBucket(kind)
+	if err != nil {
+		return nil, err
+	}
+
+	entry, err := bucket.Get(facility + "." + condID)
+	if err != nil {
+		return nil, err
+	}
+
+	return entry, nil
+}
+
+// facility is always the initial token of the key, and is terminated by a period.
+func matchFacility(facility, key string) bool {
+	pre, _, found := strings.Cut(key, ".")
+	if !found {
+		return false
+	}
+	return pre == facility
+}
+
+// GetAllConditions returns all conditions for a specific type and facility. This includes any
+// entry in any state, provided it has not been reaped by TTL.
+func GetAllConditions(kind ptypes.ConditionKind, facility string) ([]nats.KeyValueEntry, error) {
+	bucket, err := getKVBucket(kind)
+	if err != nil {
+		return nil, err
+	}
+
+	// instead of making multiple calls (one to Keys() and then multiple ones to Get(),
```

Comment:
I don't think so. This is a shortcut for "dump me the current contents of the KV" and it lives only for the duration of this function. I'm more concerned with the longer-lived watchers (cf. `startConditionWatchers()` in updates.go) because it seems like *those* can have some weird behaviors. We could consider dropping the watch functionality in its long-lived use, but I don't really like that option, because the watcher will be more responsive.

Reply:
I did make this a little more restrictive (using `Watch(facilityKey, ...)` to make this simpler and to reduce the amount of data coming back across long links (sv15 -> sg5, for example). I suppose we'll see how it works.